packages/teleport-plugin-react-base-component/src/utils.ts

```diff
@@ -483,3 +484,67 @@ const addAttributeToNode: AttributeAssignCodeMod<types.JSXElement> = (
       )
   }
 }
+
+export const generateRootNode = (
+  node: UIDLNode,
+  name: string,
+  accumulators: ReactComponentAccumulators
+) => {
+  switch (node.type) {
+    case 'static': {
+      const arrowFunction = types.arrowFunctionExpression(
+        [types.identifier('props')],
+        types.stringLiteral(node.content.toString())
+      )
+
+      const declarator = types.variableDeclarator(types.identifier(name), arrowFunction)
+
+      const component = types.variableDeclaration('const', [declarator])
+
+      return component
+    }
+
+    case 'dynamic': {
+      const memberExpression = types.memberExpression(
+        types.identifier('props'),
+        types.identifier(node.content.id)
+      )
+
+      const arrowFunction = types.arrowFunctionExpression(
+        [types.identifier('props')],
+        memberExpression
+      )
+
+      const declarator = types.variableDeclarator(types.identifier(name), arrowFunction)
+
+      const component = types.variableDeclaration('const', [declarator])
+
+      return component
+    }
+
+    case 'conditional': {
+      const expressionRightHand: types.JSXElement | types.StringLiteral =
+        typeof node.content.node.content === 'string'
+          ? types.stringLiteral(node.content.node.content.toString())
+          : generateElementNode(node.content.node as UIDLElementNode, accumulators)
+
+      const memberExpression = types.memberExpression(
+        types.identifier('props'),
+        types.identifier(node.content.reference.content.id)
+      )
+
+      const logicalExpression = types.logicalExpression('&&', memberExpression, expressionRightHand)
```

Comment:
Yeah @alexnm we can ideally use that but I did not go with them because they wrap elements using ``jsxExpressionContainers `` in our case here the structure of the output code that we are exporting, here are they are being mapped to ``arrowFunction``  and to ``variableDeclaration`` so i build a helper for these three root cases and rest of all the cases we can handle as usual.

Reply:
what if we change the `generateNodeSyntax` function to not wrap dynamic/repeat/conditional with a `jsxExpressionContainer` and do inside `addChildJSXTag`, line 81 in utils.ts. a jsxExpressionContainer should be needed only when a node is added as a child to an existing node / element right?